InstantPrintTool.py

```diff
@@ -28,9 +39,10 @@ def __init__(self, iface, populateCompositionFz=None):
         self.pressPos = None
         self.populateCompositionFz = populateCompositionFz
 
-        self.dialog = QDialog(self.iface.mainWindow())
+        self.dialog = InstantPrintDialog(parent)
```

Comment:
Please do some minimal tests whether your changes do actually work. `parent` is a non-existing variable and consequently the plugin does not even load.